### PR TITLE
fix: [CDS-45115]: Nexus logs fix

### PR DESCRIPTION
**Pull request by shivamnegi94**
## Describe your changes

- Nexus2 service logs was showing type as Nexus3Registry, fixed the issue.
- Nexus2 API support for ServiceV2
- Added the missing expression - displayName, fileName and metadata.
- Added UT for Nexus Coverage

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)

### CODE REVIEW

**Comment by shivamnegi94** [comment]
trigger ti1, ti2, ti4, ft1, ft2, smartchecks

**Comment by shivamnegi94** [comment]
trigger checkstylecodeformat

**Comment by shivamnegi94** [comment]
trigger messagecheck

**Comment by shivamnegi94** [comment]
trigger codeformat

**Review by vivekkarnatiharness — DISMISSED** [review]
> Reviewed 3 of 13 files at r1, 10 of 10 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: 13 file(s) to review. 1 of 2 approvals obtained.

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/NexusArtifactOutcome.java [hunk 1]
```diff
@@ -57,6 +57,10 @@ public class NexusArtifactOutcome implements ArtifactOutcome {
 
   String registryHostname;
 
+  String revision;
```
Comment (puthrayaharness) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/NexusArtifactOutcome.java:60]: Why are we adding this?

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/NexusArtifactOutcome.java (hunk 1 above)
Comment (shivamnegi94) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/NexusArtifactOutcome.java:60]: we have revision in CG

Comment (shivamnegi94) [reply 127-cd-nextgen-entities/src/main/java/io/harness/cdng/artifact/outcome/NexusArtifactOutcome.java:60]: To support all expression so added the revision as well

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 1 of 13 files at r1, 7 of 10 files at r2, 4 of 5 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Review by vivekkarnatiharness — DISMISSED** [review]
> Reviewed 3 of 5 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: 1 file(s) to review.

**Review by shivamnegi94 — COMMENTED** [review]
> Reviewed 1 of 13 files at r1, 7 of 10 files at r2, 3 of 5 files at r3, 4 of 4 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: 1 file(s) to review.

**Review by vivekkarnatiharness — DISMISSED** [review]
> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by puthrayaharness — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 3 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: 4 file(s) to review. 1 of 2 approvals obtained.

**Review by vtxorxwitty — APPROVED** [review]
> Reviewed 1 of 13 files at r1, 6 of 10 files at r2, 3 of 5 files at r3, 3 of 4 files at r4, 1 of 3 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: :shipit: complete! ✓

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 13 files at r1, 6 of 10 files at r2, 2 of 5 files at r3, 3 of 4 files at r4, 1 of 3 files at r5, 3 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/40690)* status: :shipit: complete! ✓

**Pull request merged**